Software/Backend/WebApi/Controllers/JobController.cs

```diff
@@ -63,4 +64,22 @@ public ActionResult<GetJobsForCurrentUserResponse> GetJobsForCurrentUser()
         return jobs;
     }
 
+    [HttpPut("CallForWorking")]
+    [Authorize]
+    public ActionResult<CallWarkerToJobResponse> CallWorkerToJob([FromBody] CallWorkerToJobRequest request)
+    {
+        var userIdFromJwt = HttpContext.Items["UserId"] as int?;
+
+        if (userIdFromJwt == null)
+        {
+            return Unauthorized(new CallWarkerToJobResponse()
+            {
+                Success = false,
+                Message = "Ne mo�ete pristupiti tom resursu!"
+            });
+        }
+        return _jobService.CallWorkerToJob(request);
+
+    }
+
```

Comment:
Ovdje se javlja jedan problem, a to je što bilo koji prijavljen korisnik može pozvati bilo kojeg radnika na bilo koji posao, a ne samo na svoj posao, jer se ne provjerava pripada li posao tom korisniku koji je prijavljen (korisnik čiji je ID jednak userIdFromJwt). Jedan od načina da se to riješi je da u CallWorkerToJobId doda polje UserId, pa u servisu ili repozitoriju ili gdje već provjeravaš je li taj Job ima za korisnika koji je stvorio taj UserId.

Iako ne znam da ovo mijenjamo ili ne, pošto koliko sam vidio na drugoj provjeri, fokus je na frontendu, a backend se ni ne treba gledati, a još je dosta timova imalo samo API pozive bez autorizacije uopće, tako da možda i ovo može proći, no ako ćemo ići dalje s ovom aplikacijom, ovo je sigurnosna rupa koju bude trebalo popraviti.